lib/auth.js

```diff
@@ -13,306 +13,18 @@
-	args || (args = {});
-	args.titaniumHomeDir = afs.resolvePath(args.titaniumHomeDir || defaultTitaniumHomeDir);
-
-	try {
-		assertSessionFile(args.titaniumHomeDir);
-	} catch (ex) {
-		args.callback(ex);
-		return;
-	}
-
-	cachedStatus = null;
-	var sessionFile = path.join(args.titaniumHomeDir, 'auth_session.json');
-
-	if (!fs.existsSync(sessionFile)) {
-		// Create a default (logged out) session file
-		args.callback(null, mix(createLoggedOutSessionFile(sessionFile), { success: true, alreadyLoggedOut: true }));
-		return;
-	}
-
-	try {
-		var session = JSON.parse(fs.readFileSync(sessionFile));
-		if (session.loggedIn) {
-			request({
-				uri: args.logoutUrl || defaultLogoutUrl,
-				method: 'GET',
-				proxy: args.proxy,
-				headers: {
-					Cookie: session.cookie
-				},
-				rejectUnauthorized: args.rejectUnauthorized === undefined ? true : !!args.rejectUnauthorized
-			}, function (error, response, body) {
-				var result = createLoggedOutSessionFile(sessionFile);
-				try {
-					if (error) {
-						throw new Error(__('Error communicating with the server: %s', error));
-					}
-
-					var res = JSON.parse(body);
-					if (res.success) {
-						mix(result, { success: true, alreadyLoggedOut: false });
-					} else {
-						throw new Error(__('Error logging out from server: %s', res.reason));
-					}
-
-					args.callback(null, result);
-				} catch (ex) {
-					args.callback(ex, result);
-				}
-			});
-		} else {
-			args.callback(null, mix(session, { success: true, alreadyLoggedOut: true }));
-		}
-	} catch (ex) { // Invalid session file. This should never happen
-		args.callback(ex, mix(createLoggedOutSessionFile(sessionFile), { success: true, alreadyLoggedOut: true }));
-	}
-};
-
-/**
- * Returns whether the user is current logged in.
- * @param {Object} [args] - Status arguments
- * @param {String} [args.titaniumHomeDir] - The Titanium home directory where the session files are stored
- * @returns {Object} An object containing the session status
- */
-exports.status = function status(args) {
-	if (cachedStatus) {
-		return cachedStatus;
-	}
-
-	args || (args = {});
-	args.titaniumHomeDir = afs.resolvePath(args.titaniumHomeDir || defaultTitaniumHomeDir);
-
-	var sessionFile = path.join(args.titaniumHomeDir, 'auth_session.json'),
-		result = {},
-		session;
-
-	if (fs.existsSync(sessionFile)) {
-		try {
-			// Fetch and parse the session data
-			session = JSON.parse(fs.readFileSync(sessionFile));
-			result = {
-				loggedIn: session.loggedIn,
-				uid: session.data && session.data.uid,
-				guid: session.data && session.data.guid,
-				email: session.data && session.data.email,
-				cookie: session.cookie
-			};
-		} catch (e) { // Invalid session file. This should never happen
-			result = createLoggedOutSessionFile(sessionFile);
-		}
-	} else {
-		result = createLoggedOutSessionFile(sessionFile); // No prior history, create a new logged out file
-	}
-
-	return cachedStatus = result;
+exports.login = function login(_args) {};
```

Comment:
Elsewhere you had to put a comment in the body of a scope to I believe make lint happy. I'm curious why linting complained elsewhere, but not here? FWIW, I do prefer the empty body as you have it.

Reply:
I prefer this too, even for the code blocks in an if or catch. We could add `allowEmptyCatch` to ignore this warning

Reply:
Yeah, I say we add `allowEmptyCatch` (which I didn't know existed!)

Reply:
done :+1: